yt/docs/en/_includes/user-guide/query-tracker/system-python-udf.md

```diff
@@ -1 +1,215 @@
+
+Example of declaring a function name and signature:
+`$f = SystemPython3_8::foo(Callable<()->String>, $script);`. In this example:
+
+- `$f` and `$script`: [Named YQL expressions](../../../yql/syntax/expressions.md#named-nodes).
+  - `$f`: Ready-to-use function.
+  - `$script`: Python script mentioned in the previous section.
+- `SystemPython3_8::`: Fixed prefix for declaring a Python function. `System` means that Python must be used from the environment at runtime. `3_8` indicates the Python version to be used.
+  - As of the time of writing this article, versions `3_8` – `3_12` are available.
+- `foo`: Name of the function to be called, defined in the `$script`.
+- `Callable<()->String>`: Function signature description. The empty brackets indicate that the function doesn't accept any arguments. `String` after the arrow means that the function returns a string. Example of a function signature with arguments: `Callable<(String, Uint32)->Double>`.
+
+### Use UDF in your query
+
+You can call the function obtained after declaring the signature in the same way as any other built-in YQL function. For example, `$f()` or `$udf("test", 123)`.
+
+## Data types
+
+To find out which data types are allowed in the function signature, see the [YQL data types](../../../yql/types/index.md) section.
+
+### Containers
+
+Containers are converted into Python objects according to the following rules:
+
+| **Name** | **Signature declaration** | **Signature example** | **Representation in Python** |
+| --- | --- | --- | --- |
+| List | `List<Type>` | `List<Int32>` | `list-like object` [(learn more)](#python-container-wrappers) |
+| Dictionary | `Dict<KeyType,ValueType>` | `Dict<String,Int32>` | `dict-like object` [(learn more)](#python-container-wrappers) |
+| Tuple | `Tuple<Type1,...,TypeN>` | `Tuple<Int32,Int32>` | `tuple` |
+| Structure | `Struct<Name1:Type1,...,NameN:TypeN>` | `Struct<Name:String,Age:Int32>` | `StructSequence` |
+| Stream | `Stream<Type>` | `Stream<Int32>` | `generator` |
+| Variant on tuple | `Variant<Type1,Type2>` | `Variant<Int32,String>` | `tuple with index and object` |
+| Variant on structure | `Variant<Name1:Type1,Name2:Type2>` | `Variant<value:Int32,error:String>` | `tuple with field name and object` |
+
+You can nest containers into each other. For example, `List<Tuple<Int32,Int32>>`.
+
+### Specifics of objects for lists and dictionaries passed to functions {#python-container-wrappers}
+
+When you use `List` and `Dict`, the read-only `yql.TList` and `yql.TDict` objects are passed to the function arguments.
+
+The objects' specifics:
+
+- They are read-only. This means that you can't make any changes to these objects without copying them because another part of the current query, for example, a neighboring function, may use the same object.
```

Comment:
```suggestion
- They are read-only. This means that you can't make any changes to these objects without copying them because another part of the current query may use the same object. For example, a neighboring function.
```